### DISCUSSION TITLE
extract_tables() returning empty with explicit vertical lines

### DISCUSSION THREAD

**Discussion by juliamyndshft** [jsvine/pdfplumber#737] [Ask for help with specific PDFs]
Hi, I'm having an issue with this pdf:
[pa-matrix-2018.pdf](https://github.com/jsvine/pdfplumber/files/9727471/pa-matrix-2018.pdf)

I'm using this code:

```
bounding_box = (0, 600, 900, 2455)
ts = {"vertical_strategy": "explicit",
         "horizontal_strategy": "text",
         "explicit_vertical_lines": "[150, 200, 850, 900]"}
cropped_page = pdf.pages[0].crop(bounding_box)
table = cropped_page.extract_tables(table_settings=ts)
```

I'm pretty sure the vertical lines are correct for grabbing the first and fourth columns on the page, but the table comes up empty. It's worth noting that when I use extract_text(), all the text in the document shows up fine, but I need it to be separated by column. What could be happening here?

Thanks!
Julia

**Comment by jsvine** [comment]
Hi @juliamyndshft, and thanks for your interest in this library. One issue I see is the quoting of the list in `explicit_vertical_lines` — instead of  `"[150, 200, 850, 900]"`, try `[150, 200, 850, 900]`. Does that fix things?

**Reply by juliamyndshft** [reply]
Hi @jsvine, thanks for your response! Apologies, that was a typo on my part; it's without the quotes in my code, so I do get the same result.

**Reply by jsvine** [reply]
Ah, looks like the lines might not quite be where you wanted them:

```python
im = cropped_page.to_image()
im.reset().draw_vlines(ts["explicit_vertical_lines"])
```

...produces this:

![image](https://user-images.githubusercontent.com/534702/194397956-00c0cbbc-dcb4-4136-b7c5-f0a93533baaa.png)

I don't know what process you used to select the line positions, but one easy way to trial-and-error it is to use use the `PageImage.draw_*(...)` methods documented [here](https://github.com/jsvine/pdfplumber#drawing-methods).

**Reply by jsvine** [reply]
One other thing that will be handy is using `PageImage.debug_tablefinder(...)`. E.g., the current settings:

```python
im.reset().debug_tablefinder(ts)
```
... produce this:

![image](https://user-images.githubusercontent.com/534702/194398698-d70282fb-4074-4d08-83e5-738c762b74e2.png)

**Reply by juliamyndshft** [reply]
Ah! I didn't realize the lines needed to be so tight on the edges. This worked with the explicit vertical lines of [175, 210, 850, 900]. Thanks so much!
